app/src/main/java/io/mercury/coroutinesandbox/view/downloader/DownloadListActivity.kt

```diff
@@ -0,0 +1,131 @@
+
+@AndroidEntryPoint
+class DownloadListActivity : ComponentActivity() {
+
+    @Inject
+    lateinit var downloadMovie: DownloadMovie
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val model: DownloadListViewModel by viewModels()
+
+        setContent {
+            val composeState by model.feature.state.collectAsState(lifecycleScope.coroutineContext)
+            Root(composeState)
+        }
+
+        model.feature.load()
+    }
+
+    @Composable
+    fun Root(state: State) {
+        ThemedMaterial {
+            Box(
+                Modifier
+                    .background(MaterialTheme.colors.background)
+                    .fillMaxSize()
+                    .padding(16.dp)
+            ) {
+                when (state) {
+                    is Uninitialized,
+                    is Loading -> Text("Loading...")
+                    is Error -> Text("ERROR")
+                    is Loaded -> {
+                        Column(
+                            verticalArrangement = Arrangement.spacedBy(24.dp)
+                        ) {
+                            Column(
+                                modifier = Modifier
+                                    .weight(1f)
+                                    .background(Color.LightGray)
+                                    .fillMaxWidth()
+                            ) {
+                                Text("Downloading:")
+                                LazyColumn {
+                                    items(state.inProgressDownloads) { download ->
+                                        Text("${download.title} - ${download.downloadProgress}%")
+                                    }
+                                }
+                            }
+
+                            Column(
+                                modifier = Modifier
+                                    .weight(1f)
+                                    .background(Color.Cyan)
+                                    .fillMaxWidth()
+                            ) {
+                                Text("Finished:")
+                                LazyColumn {
+                                    items(state.finishedDownloads) { download ->
+                                        Text(download.title)
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+
+                Button(
+                    onClick = ::downloadSomething,
+                    modifier = Modifier.align(Alignment.BottomEnd)
+                ) {
+                    Text("Download Something")
+
+                }
+            }
+        }
+    }
+
+    private val downloadables = Stack<String>().also {
+        it.push("tt0068646")
+        it.push("tt0111161")
+        it.push("tt0108052")
+    }
+
+    private fun downloadSomething() {
```

Comment:
I feel like this belongs in the `DownloadListViewModel`. The views should be as dumb as possible imo.

Reply:
This was me just being lazy and putting this here just to have a way to kick off the faux downloads.

This brings up a question I have had... While I 100% agree w/ moving the logic out of the view layer (Activity in this case), I am wondering if it makes sense to move it into the existing feature. My thoughts are that it doesn't *directly* affect the state of the DownloadListFeature, so maybe it doesn't belong there. This might pair well with the smart component approach as it would make more sense to keep that state specific to the button, and not the entire screen.